### DISCUSSION TITLE
zlib-ng and zlibwapi.dll

### DISCUSSION THREAD

**Discussion by Spatzendame** [zlib-ng/zlib-ng#1180] [Q&A]
Hello, I want to replace the zlib1.dll of a video game with the more modern and more optimized zlib-ng. But the game does also use the zlibwapi.dll in combination with zlib1.dll. So does nlib-ng work together with zlibwapi.dll?

Thank you for your answer.

**Comment by mtl1979** [comment]
Most likely zlibwapi.dll needs to be recompiled... Without seeing the symbols it exported, it's hard to say what it actually does... It might be just normal zlib compiled with different calling convention (Windows API).

**Reply by mgorny** [reply]
Yes, `zlibwapi.dll` is just zlib built with `-DZLIB_WINAPI`, see e.g. https://github.com/conda-forge/zlib-feedstock/blob/3240862e6d0edaa8ec1e624301628c8056950ba0/recipe/bld.bat#L18-L19

**Reply by mtl1979** [reply]
I have never tried to test if ZLIB-NG is API compatible with stock zlib when compiled for Windows API calling convention. There are known compatibility issues with 64-bit Windows, but not any known issues with 32-bit Windows as people don't use it often anymore... I have one PC running Windows 98 SE but it does not have compatible version of Visual C++ installed. It is possible to install Visual C++ 2017.9 on 32-bit Windows but it has unfixed bugs and it's out of support already.

**Comment by KungFuJesus** [comment]
One unfortunate side effect of it being a windows game is that it's likely stuck as a 32 bit binary where your register file is cut in half.  We still work as a 32 bit binary but it does leave some performance on the table, particularly in register starved scenarios.
